internal/core/src/segcore/SegmentSealedImpl.cpp

```diff
@@ -912,6 +912,17 @@ SegmentSealedImpl::check_search(const query::Plan* plan) const {
     AssertInfo(plan->extra_info_opt_.has_value(),
                "Extra info of search plan doesn't have value");
 
+    auto& metric_str = plan->plan_node_->search_info_.metric_type_;
```

Comment:
Is it possible to abstract the above code into a function or place it in a base class?